pallets/erc20-app/src/lib.rs

```diff
@@ -92,11 +96,23 @@ decl_module! {
                 return Err(Error::<T>::InvalidAssetId.into())
             }
 
-            // <asset::Module<T>>::do_burn(asset_id, &who, amount)?;
+
             let native_asset_id = <asset::Module<T>>::get_native_asset_id(asset_id);
-            let result = <assets::Module<T>>::assets_burn(&native_asset_id, &who, &amount.low_u128().saturated_into::<T::Balance>());
 
-            ensure!(result.is_ok(), Error::<T>::BurnFailure);
+            ensure!(T::Currency::can_slash(
```

Comment:
I do think we can use slash instead of burn here, because are checking can_slash which checks for free balance availability.
Still maybe we should check with Gleb once.
Without can_slash check it might actually slash from reserved balance and might not even slash the entire "burn" amount if free+reserved do not suffice.

Reply:
I think we also need to check ensure_can_withdraw, as I believe that when burning frozen amount should be considered.